dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1665,6 +1674,46 @@ protected synchronized void writeStreamOutput(SystemLogHolder holder, String txt
         holder.originalStream.println(txt);
     }
 
+    /**
+     * This method is accessed via reflection by the com.ibm.ws.logging.osgi.stackjoiner.bci.ThrowableInfo class
+     * It will be called on any Throwable.printStackTrace(PrintStream) invocation.
+     *
+     * @param t reference to the current Throwable object calling printStackTrace
+     * @param originalStream reference to the PrintStream object to be written to
+     * @return true if the printStackTrace method was overridden, false otherwise
+     */
+    public static boolean printStackTraceOverride(Throwable t, PrintStream originalStream) {
```

Comment:
@jhanders34 by calling `t.printStackTrace(originalStream);` we invoke the original logic for formatting the Throwable stack, which varies depending on whether you are writing to console (where we omit internal classes) or to messages.log (where we show all stack frames).  The logic of determining which stack frames to omit was difficult to extract into this printStackTraceOverride method since it has a lot of dependencies.

Reply:
I was saying to have a custom PrintWriter to call the filter logic.  I didn't think the logic had that many dependencies, but you know the code more than I do.

Reply:
To be clear I was not thinking to bring over entire classes, but just methods that were needed.  Those methods would move to ThrowableProxy or some other class in the same place and would just be referenced from all places where it is currently used.

Reply:
... I like your suggestion that everything could use the ThrowableProxy, but the other challenge then would be the visibility between the classes that currently use that and the classloader that has ThrowableProxy.